src/chart/components/chart/chart.component.ts

```diff
@@ -228,7 +228,8 @@ export class ChartComponent extends ChartBaseElement {
 			new MainChartHistogramDrawer(this.config.components.chart.histogram),
 		);
 
-		const mainChartBoundsProvider = () => this.canvasBoundsContainer.getBounds(CanvasElement.PANE_UUID(CHART_UUID));
+		const chartPaneId = CanvasElement.PANE_UUID(CHART_UUID);
+		const mainChartBoundsProvider = () => this.canvasBoundsContainer.getBounds(chartPaneId);
```

Comment:
have you tried to remove "provider" logic at all and pass bounds as is without function? I think it might work since bounds is mutable and new instance shouldn't be created